### PR TITLE
refactor: truncate support build pipeline

### PR DESCRIPTION
**Pull request by zhyass**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

1. truncate add table lock.
2. truncate support build pipeline. 
    OneBlockSource --------> CommitSink

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [x] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

### CODE REVIEW

File: src/query/service/src/interpreters/interpreter_delete.rs [hunk 1]
```diff
@@ -197,48 +198,93 @@ impl Interpreter for DeleteInterpreter {
...
         let mut build_res = PipelineBuildResult::create();
+
+        let snapshot_opt = fuse_table.read_table_snapshot().await?;
+        // check if table is empty
+        let snapshot = if let Some(val) = snapshot_opt {
+            val
+        } else {
+            // no snapshot, no deletion
+            return Ok(build_res);
+        };
```
Comment (SkyFan2002) [inline src/query/service/src/interpreters/interpreter_delete.rs:209]:
```suggestion
        let Some(snapshot) = fuse_table.read_table_snapshot().await? else {
            // no snapshot, no deletion
            return Ok(build_res);
        };
```

File: src/query/storages/fuse/src/operations/common/processors/sink_commit.rs [hunk 2]
```diff
@@ -342,13 +349,22 @@ where F: SnapshotGenerator + Send + 'static
                 .await
                 {
                     Ok(_) => {
-                        if self.transient {
-                            // Removes historical data, if table is transient
+                        if Self::is_truncate() {
+                            catalog
+                                .truncate_table(&table_info, TruncateTableReq {
```
Comment (SkyFan2002) [inline src/query/storages/fuse/src/operations/common/processors/sink_commit.rs:354]: Maybe `catalog.truncate_table()` should not be called when in the delete fast path.

Comment (zhyass) [reply src/query/storages/fuse/src/operations/common/processors/sink_commit.rs:370]: > Maybe `catalog.truncate_table()` should not be called when in the delete fast path. Delete stmt should only delete the table data, not influence the `copied_files` in meta.

Thx, fixed

File: src/query/service/src/interpreters/interpreter_delete.rs (hunk 1 above)
Comment (zhyass) [reply src/query/service/src/interpreters/interpreter_delete.rs]: Thx, get a new technique!

**Review by SkyFan2002 — APPROVED** [review]

**Pull request merged**